src/cmd/build.js

```diff
@@ -95,11 +95,23 @@ export function safeFileName(name: string): string {
 export class FileFilter {
   filesToIgnore: Array<string>;
 
-  constructor({filesToIgnore}: Object = {}) {
+  constructor(artifactsDir,{filesToIgnore}: Object = {}) {
+    var eliminateArtifactDir =  new String();
+    var buf = artifactsDir;
+    if(typeof buf !== 'undefined' && buf.indexOf('web-ext-artifacts') != -1){
+        eliminateArtifactDir = buf.slice(buf.indexOf('web-ext-artifacts'));
+    }
+    else if(typeof buf !== 'undefined' && buf.slice(-1) === '/'){
+	eliminateArtifactDir = path.join(buf.slice(0,-1));
+    }
+    else if(typeof buf !== 'undefined'){
+	eliminateArtifactDir =  path.join(buf);
+    }
     this.filesToIgnore = filesToIgnore || [
       '**/*.xpi',
       '**/*.zip',
       '**/.*', // any hidden file
+      '**/*'+eliminateArtifactDir,
```

Comment:
I don't like the complexity in your current patch, there are too many steps taken to convert the path to a minimatch value. If we introduce another for-loop I don't think it will pose a performance problem. The minimatch pattern matcher is probably slower than a straight path check anyway.